website/source/docs/secrets/ad/index.html.md

```diff
@@ -132,14 +138,202 @@ The password `ttl` on a role can be updated at any time to ensure that the
+service_account_name    buzz@example.com
+```
+
+This can be a nice way to say, "Although I _can_ have a check-out for 24 hours, if I 
+haven't checked it in after 30 minutes, I forgot or I'm a dead instance, so you can just
+check it back in."
+
+If no service accounts are available for check-out, Vault will return a 400 Bad Request.
+
+```text
+$ vault write -f ad/library/accounting-team/check-out
+Error writing data to ad/library/accounting-team/check-out: Error making API request.
+
+URL: PUT http://localhost:8200/v1/ad/library/accounting-team/check-out
+Code: 400. Errors:
+
+* No service accounts available for check-out.
+```
+
+To extend a check-out, renew its lease.
+
+```text
+$ vault lease renew ad/library/accounting-team/check-out/0C2wmeaDmsToVFc0zDiX9cMq
+Key                Value
+---                -----
+lease_id           ad/library/accounting-team/check-out/0C2wmeaDmsToVFc0zDiX9cMq
+lease_duration     10h
+lease_renewable    true
+```
+
+Renewing a check-out means its current password will live longer, since passwords are rotated 
+anytime a password is _checked in_ either by a caller, or by Vault because the check-out `ttl`
+ends.
+
+To check a service account back in for others to use, call:
+
+```text
+$ vault write -f ad/library/accounting-team/check-in
+Key          Value
+---          -----
+check_ins    [fizz@example.com]
+```
+
+Most of the time this will just work, but if multiple service accounts checked out by the same 
+caller, Vault will need to know which one(s) to check in.
+
+```text
+$ vault write ad/library/accounting-team/check-in service_account_names=fizz@example.com
+Key          Value
+---          -----
+check_ins    [fizz@example.com]
+```
+
+To perform a check-in, Vault verifies that the caller _should_ be able to check in a given service account.
+To do this, Vault looks for either the same [entity ID](https://learn.hashicorp.com/vault/identity-access-management/iam-identity) 
+used to check out the service account, or the same client token.
+
+If a caller is unable to check in a service account, or simply doesn't try,
+Vault will check it back in automatically when the `ttl` expires. However, if that is too long, 
+service accounts can be forcibly checked in by a highly privileged user through:
+
+```text
+$ vault write -f ad/library/manage/accounting-team/check-in
+Key          Value
+---          -----
+check_ins    [fizz@example.com]
+```
+
+Or, alternatively, revoking the secret's lease has the same effect.
+
+```text
+$ vault lease revoke ad/library/accounting-team/check-out/PvBVG0m7pEg2940Cb3Jw3KpJ
+All revocation operations queued successfully!
+```
+
+### Troubleshooting
+
+#### Old passwords are still valid for 5 minutes.
+
+During testing, we found that by default, many versions of Active Directory
+perpetuate old passwords for 5 minutes. After we discovered this behavior,
+we found articles discussing it by searching for "AD password caching". We 
```

Comment:
I'd also add to the search term suggestions: "OldPasswordAllowedPeriod" (a registry setting), since that reg entry appears widely on this topic and is a nice specific, searchable term.